.github/workflows/release-cycle.yml

```
@@ -0,0 +1,34 @@
+name: Test release cycle
+
+on: [pull_request, push, workflow_dispatch]
+
+env:
+  FORCE_COLOR: 1
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Silence Windows warnings
+        run: |
+          git config core.eol lf
+          git config core.autocrlf true
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3"
+
+      - name: Generate release cycle output
+        run: python _tools/generate_release_cycle.py
```

For a more robust test run, I suggest this standard invocation:

```suggestion
        run: python -I -bb -X dev -X warn_default_encoding -W error _tools/generate_release_cycle.py
```

This will run in isolated mode, catch btyes-string issues, turn on warnings (e.g. deprecationwarnings, resourcewarnings, etc) and make them errors as well as a few other checks, and also warn on the implicit default encoding (which would have caught the various `open` calls missing `encoding` parameters).